driver/asana_driver.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+
+# stdlib imports
+import os
+import pprint
+# third-party imports
+import asana
+import dateparser
+import pytz
+import requests
+# local imports
+from lib import AuthRevokedError
+from lib import RateLimitError
+from lib import RetrieveDataResult
+from lib import RetrieveMetadataResult
+from lib import ServiceUnavailableError
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+
+def parse_date(s):
+    if not s: return None
+    
+    dt = dateparser.parse(s)
+    dt = pytz.utc.normalize(dt.astimezone(pytz.utc))
+    
+    return dt
+
+
+class AsanaDriver(object):
+
+    def __init__(self, butter_user_id, datasource_user_id):
+        self.butter_user_id = butter_user_id
+        self.datasource_user_id = datasource_user_id
+        
+        # pull api credentials
+        personal_access_token = os.environ['ASANA_ACCESS_TOKEN'] # TODO: handle authentication better
+        
+        # create api client
+        self.client = asana.Client.access_token(personal_access_token)
+        
+        # pull initial data
```

Comment:
Let's move this into another function, such that we can create a driver object where the initialization does as little work as possible. For example, it looks like `retrieve_data` isn't relying on any of the network setup here -- it'd be wasted IO in that scenario.